### PR TITLE
EntitasFF updated to Dart 3! Added new null safety methods.

### PR DESCRIPTION
**Pull request by chqs-git**
Pull request to update project to the newest Dart version.

### CODE REVIEW

File: lib/src/state.dart [hunk 1]
```diff
@@ -484,17 +494,31 @@ class EntityManager implements EntityObserver {
 
   /// Returns the component instance or `null`.
```
Comment (mzaks) [inline lib/src/state.dart]: please fix the comment, it should be `Returns the component instance or throws an exception if component is null`

File: lib/src/state.dart [hunk 2]
```diff
@@ -484,17 +494,31 @@ class EntityManager implements EntityObserver {
...
+    return component;
   }
 
-  /// Returns [Entity] instance which hold the unique component, or `null`.
+  /// Returns the component instance or `null`.
+  T? getUniqueOrNull<T extends UniqueComponent>() {
+    return _uniqueEntities[T]?.getOrNull<T>();
+  }
+
+  /// Returns [Entity] instance which hold the unique component
```
Comment (mzaks) [inline lib/src/state.dart]: please mention that the method throws when entity is null

**Review by mzaks — COMMENTED** [review]
Great PR, thank you so much!

Please address two small things I found regarding the API documentation.
When it is done I will approve and merge.

**Comment by chqs-git** [comment]
Great, yeah I will change it in a bit thank you.

A segunda, 22/04/2024, 16:15, Maxim Zaks ***@***.***>
escreveu:

**Comment by chqs-git** [comment]
Hopefully everything is alright now :).
Cheers.

**Review by mzaks — APPROVED** [review]
Looks great! Thank you!

**Pull request merged**